ttnn/cpp/ttnn/operations/pool/downsample/downsample_op_pybind.hpp

```diff
@@ -32,10 +32,10 @@ void bind_downsample(py::module& module) {
         ttnn::downsample,
         doc,
         ttnn::pybind_arguments_t{
-            py::arg("input_tensor"),
```

Comment:
rename file to downsample_pybind and extract implementation to .cpp

For more context see this message and example PRs
https://tenstorrent.slack.com/archives/C065NUB85K3/p1722289334151969

Reply:
Best example of pybind header is here
https://github.com/tenstorrent/tt-metal/blob/main/ttnn/cpp/ttnn/operations/ccl/all_gather/all_gather_pybind.hpp

Reply:
Done. @ayerofieiev-tt  Only thing is in downsample.hpp I have changed register_operation from register_operation_with_auto_launch_op. Not well versed in that part of framework. pls do let me know if undoing is required.

Reply:
@nkpatel-tt yes please use register_operation like we discussed.